game/transfers.py

```diff
@@ -659,6 +666,7 @@ def cancel_transfer(self, transfer: TransferOrder) -> None:
             self.cancel_transport(transfer.transport, transfer)
         self.pending_transfers.remove(transfer)
         transfer.origin.base.commission_units(transfer.units)
+        self._update_eventstream()
```

Comment:
Where did you land on whether this was needed or not? If this is still needed then we're good, otherwise replace this with a comment like:

```
# Transfers are only ever split by airlifts, and airlifts do not travel along supply routes, so no update is needed.
```

and rename the function `_send_supply_route_event_stream_update`

Reply:
Split transfer doesn't require an update, but where exactly do you want that comment?

Reply:
Also, you do realize that you're commenting on the update that's happening in `cancel_transfer` right?

Reviewing this I'm starting to doubt if the update is really not needed in `split_transfer`, because I may have missed the case where the last possible airlift is planned, meaning that the subsequent split will force the remaining units to take the road. I'll try investigating this theory but if it turns out that it updates properly through a different path then I don't think we'll need to push updates from `split_transfer`.

Reply:
After closer inspection I'm starting to think it's actually necessary to push updates as well from `split_transfer`, and that doesn't seem to be the only push that's needed. I noticed transport flights not being drawn until you try editing the package or flight, causing updates to be sent which draws the flight plan. I also noticed the `AirliftPlanner` is simply ignoring C130s for transport. I tried transferring units from an airbase that has C130s to another airbase, which in theory should work fine. Helicopters work (except for not updating the front-end properly), but I'm pretty sure cargo planes should work too.
The one thing that makes me doubt whether we should push supply route updates in `split_transfer` is the fact that nothing happens with units that don't have transport. In other words, if the last airlift was planned and there's units left to transport, they won't automatically take the road so supply routes don't require an update. However the question remains whether this is correct behavior or not. Should the remaining transfer order be cancelled in that case? Should it automatically adjust so the remaining units take the road? I'm not even sure how I would want it... -_-

Reply:
I think `split_transfer` is best left alone for the time being as there seem to be multiple things going on here. A separate PR (or more) addressing this issue is perhaps more appropriate.